### PR TITLE
Meta/connecting phylogeny back to front end

### PR DESCRIPTION
**Pull request by xcwfwf44**
I have connected the phylogeny tree in the backend to the frontend. Specifically I made sure that the marshalling of the Phylogeny classes to JSON is done in such way that the JSON produced by the server is in a format expected by the frontend. At the same time I made sure that the marshalling logic does not affect the code in the backend by using XmlAdapter classes. These classes are used by our marshaller Moxy to make the backend classes compatible for marshalling and allow for formatting the resulting JSON. I also added a test for the getphylogenetic tree REST endpoint.

### CODE REVIEW

File: tagcwebapp/src/main/java/com/pl/tagc/tagcwebapp/PhylogeneticNodeAdapter.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+public class PhylogeneticNodeAdapter extends XmlAdapter<AdaptedPhylogeneticNode, PhylogeneticNode> {
+
+	@Override
+	public PhylogeneticNode unmarshal(AdaptedPhylogeneticNode adaptedPhylogeneticNode)
+			throws Exception {
+		return null;
+	}
+
+	@Override
+	public AdaptedPhylogeneticNode marshal(PhylogeneticNode node) throws Exception {
```
Comment (Icety) [inline tagcwebapp/src/main/java/com/pl/tagc/tagcwebapp/PhylogeneticNodeAdapter.java]: Missing JavaDoc

File: tagcwebapp/src/main/resources/static/external/phylotree/phylotree.js [hunk 2]
```diff
@@ -579,21 +579,32 @@ d3.layout.phylotree = function(container) {
         return phylotree;
     };
 
-    function phylotree(nwk, bootstrap_values) {
-
+    function phylotree(tree_data, isJSONtree, bootstrap_values) {
```
Comment (Icety) [inline tagcwebapp/src/main/resources/static/external/phylotree/phylotree.js]: Add JavascriptDoc

File: tagcwebapp/src/main/resources/static/js/phylotreewidget.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 var color_scheme = d3.scale.category10();
-var cacheNewickString;
+var cached_json_data;
```
Comment (Icety) [inline tagcwebapp/src/main/resources/static/js/phylotreewidget.js]: Make this CamelCase

File: tagcwebapp/src/test/java/com/pl/tagc/tagcwebapp/RestApiTest.java [hunk 4]
```diff
@@ -1,34 +1,51 @@
...
-     * Test, that the resource response is in JSON format.
-     */
-    @Test
-    public void testRestApiResource() {
-        final Response response = target().path("api/getdimensions").request().get(Response.class);
-        assertEquals("application", response.getMediaType().getType());
-        assertEquals("json", response.getMediaType().getSubtype());
-    }
+	@Override
+	protected ResourceConfig configure() {
```
Comment (Icety) [inline tagcwebapp/src/test/java/com/pl/tagc/tagcwebapp/RestApiTest.java]: Missing JavaDoc

File: tagcwebapp/src/test/java/com/pl/tagc/tagcwebapp/RestApiTest.java [hunk 5]
```diff
@@ -1,34 +1,51 @@
...
+	 */
+	@Test
+	public void testRestApiResource() {
+		final Response response = target().path("api/getdimensions").request().get(Response.class);
+		assertEquals("application", response.getMediaType().getType());
+		assertEquals("json", response.getMediaType().getSubtype());
+	}
+
+	@Test
+	public void testGetPhylogeneticTree() throws Exception {
```
Comment (Icety) [inline tagcwebapp/src/test/java/com/pl/tagc/tagcwebapp/RestApiTest.java]: Missing JavaDoc

File: tagcwebapp/src/main/resources/static/external/phylotree/phylotree.js (hunk 2 above)
Comment (xcwfwf44) [inline tagcwebapp/src/main/resources/static/external/phylotree/phylotree.js]: thats a js library i simply made some modfication to the exisiting code

**Comment by Icety** [comment]
Looks quite good, can u maybe run over all code and let your IDE re-indent all code. At some spots there are 4 spaces/1 tab, at some 8 spaces/2 tabs. And add some Java(script)Doc.
After that it is fine to me.

File: tagcwebapp/src/main/resources/static/external/phylotree/phylotree.js (hunk 2 above)
Comment (Icety) [inline tagcwebapp/src/main/resources/static/external/phylotree/phylotree.js]: Ahh, I see, my bad

**Comment by xcwfwf44** [comment]
I think that our IDE's have different formatting rules. We should first all make sure we use the same formatting rules in our IDE's.

**Pull request merged**